### PR TITLE
ceph-config: introduce dedicated cluster config flow

### PR DESCRIPTION
**Pull request by clwluvw**

### CODE REVIEW

File: roles/ceph-config/templates/ceph.conf.j2 [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
-#{% if not cephx | bool %}
-#auth cluster required = none
-#auth service required = none
-#auth client required = none
-#{% endif %}
+{% if not cephx | bool %}
+auth cluster required = none
+auth service required = none
+auth client required = none
+{% endif %}
```
Comment (guits) [inline roles/ceph-config/templates/ceph.conf.j2:9]: why do we keep these lines?

File: site-container.yml.sample [hunk 2]
```diff
@@ -175,58 +175,25 @@
...
-        option: "{{ item.key }}"
-        value: "{{ item.value }}"
-      environment:
-        CEPH_CONTAINER_IMAGE: "{{ ceph_docker_registry + '/' + ceph_docker_image + ':' + ceph_docker_image_tag if containerized_deployment | bool else None }}"
-        CEPH_CONTAINER_BINARY: "{{ container_binary }}"
-      when: inventory_hostname == ansible_play_hosts_all | last
-      with_dict: "{{ ceph_conf_overrides['global'] }}"
-
-    - name: set osd_memory_target
+    - name: Set clsuter configs
```
Comment (guits) [inline site-container.yml.sample:186]:
```suggestion
    - name: Set cluster configs
```

File: group_vars/all.yml.sample [hunk 3]
```diff
@@ -273,6 +273,16 @@ dummy:
...
 
+# Cluster configuration
+#ceph_cluster_conf:
+#  global:
+#    public_network: "{{ public_network | default(omit) }}"
+#    cluster_network: "{{ cluster_network | default(omit) }}"
+#    osd_pool_default_crush_rule: "{{ osd_pool_default_crush_rule }}"
+#    ms_bind_ipv6: "{{ (ip_version == 'ipv6') | string }}"
+#    ms_bind_ipv4: "{{ (ip_version == 'ipv4') | string }}"
+#    osd_crush_chooseleaf_type: "{{ '0' if common_single_host_mode | default(false) else omit }}"
```
Comment (guits) [inline group_vars/all.yml.sample:284]:
```suggestion
# ceph_cluster_conf:
#   global:
#     public_network: "{{ public_network | default(omit) }}"
#     cluster_network: "{{ cluster_network | default(omit) }}"
#     osd_pool_default_crush_rule: "{{ osd_pool_default_crush_rule }}"
#     ms_bind_ipv6: "{{ (ip_version == 'ipv6') | string }}"
#     ms_bind_ipv4: "{{ (ip_version == 'ipv4') | string }}"
#     osd_crush_chooseleaf_type: "{{ '0' if common_single_host_mode | default(false) else omit }}"
```

File: group_vars/rhcs.yml.sample [hunk 4]
```diff
@@ -273,6 +273,16 @@ ceph_iscsi_config_dev: false
...
+# Cluster configuration
+#ceph_cluster_conf:
+#  global:
+#    public_network: "{{ public_network | default(omit) }}"
+#    cluster_network: "{{ cluster_network | default(omit) }}"
+#    osd_pool_default_crush_rule: "{{ osd_pool_default_crush_rule }}"
+#    ms_bind_ipv6: "{{ (ip_version == 'ipv6') | string }}"
+#    ms_bind_ipv4: "{{ (ip_version == 'ipv4') | string }}"
+#    osd_crush_chooseleaf_type: "{{ '0' if common_single_host_mode | default(false) else omit }}"
+
```
Comment (guits) [inline group_vars/rhcs.yml.sample:285]: rhcs.yml.sample should probably be dropped

File: site.yml.sample [hunk 5]
```diff
@@ -167,49 +167,22 @@
...
-      ceph_config:
-        action: set
-        who: "global"
-        option: "{{ item.key }}"
-        value: "{{ item.value }}"
-      when: inventory_hostname == ansible_play_hosts_all | last
-      with_dict: "{{ ceph_conf_overrides['global'] }}"
-
-    - name: set osd_memory_target
+    - name: Set clsuter configs
```
Comment (guits) [inline site.yml.sample:178]:
```suggestion
    - name: Set cluster configs
```

File: group_vars/all.yml.sample (hunk 3 above)
Comment (clwluvw) [reply group_vars/all.yml.sample:284]: This is generated by the script - would it work by changing it? I guess CI would complain?

**Comment by clwluvw** [comment]
kenkins test centos-container-all_in_one

**Comment by clwluvw** [comment]
jenkins test centos-non_container-all_in_one

**Comment by clwluvw** [comment]
jenkins test centos-container-all_in_one

**Comment by clwluvw** [comment]
jenkins test centos-container-all_in_one

File: group_vars/all.yml.sample (hunk 3 above)
Comment (guits) [reply group_vars/all.yml.sample:284]: We have to modify the script then 🙂

**Comment by clwluvw** [comment]
jenkins test centos-container-all_in_one

File: group_vars/all.yml.sample (hunk 3 above)
Comment (clwluvw) [reply group_vars/all.yml.sample:284]: I guess it was good to have comments for vars with no space so the real comments for vars could be distinguished from those. What do you think?

**Comment by guits** [comment]
jenkins test centos-non_container-switch_to_containers

**Comment by guits** [comment]
jenkins test centos-non_container-collocation

**Comment by guits** [comment]
jenkins test centos-non_container-collocation

**Comment by guits** [comment]
jenkins test centos-container-all_in_one

**Comment by guits** [comment]
jenkins test centos-non_container-all_in_one

**Comment by clwluvw** [comment]
jenkins test centos-non_container-all_in_one

**Comment by clwluvw** [comment]
jenkins test centos-container-lvm_batch

**Comment by clwluvw** [comment]
jenkins test centos-non_container-all_in_one

File: roles/ceph-handler/templates/restart_rgw_daemon.sh.j2 [hunk 6]
```diff
@@ -74,7 +74,16 @@ check_rest() {
 
 for ((i=0; i<${RGW_NUMS}; i++)); do
   # First, restart the daemon
-  systemctl restart ceph-radosgw@rgw.${HOST_NAME}.${INSTANCES_NAME[i]}
+
+  # Check if systemd unit exists
+  # This is needed for new instances are the restart might trigger before the deployment
```
Comment (clwluvw) [inline roles/ceph-handler/templates/restart_rgw_daemon.sh.j2:79]:
```suggestion
  # This is needed for new instances as the restart might trigger before the deployment
```

File: group_vars/rhcs.yml.sample (hunk 4 above)
Comment (clwluvw) [reply group_vars/rhcs.yml.sample]: https://github.com/ceph/ceph-ansible/commit/1af387621dfdd59373a61cb38665a3553f4a96ed

**Review by guits — APPROVED** [review]

**Pull request merged**